features/dd-sdk-android-profiling/src/test/kotlin/com/datadog/android/profiling/internal/ProfilingStorageTest.kt

```diff
@@ -0,0 +1,124 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.profiling.internal
+
+import android.content.Context
+import android.content.SharedPreferences
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@Extensions(
+    ExtendWith(MockitoExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+internal class ProfilingStorageTest {
+
+    @Mock
+    lateinit var mockContext: Context
+
+    @Mock
+    lateinit var mockPrefs: SharedPreferences
+
+    @Mock
+    lateinit var mockEditor: SharedPreferences.Editor
+
+    @BeforeEach
+    fun `set up`() {
+        whenever(mockContext.getSharedPreferences(any(), any())) doReturn mockPrefs
+        whenever(mockPrefs.edit()) doReturn mockEditor
+        whenever(mockEditor.remove(any())) doReturn mockEditor
+        whenever(mockEditor.putBoolean(any(), any())) doReturn mockEditor
+
+        // Reset the singleton
+        val storageField = ProfilingStorage::class.java.getDeclaredField("sharedPreferencesStorage")
+        storageField.isAccessible = true
+        storageField.set(ProfilingStorage, null)
```

Comment:
minor: probably it is better to do it in the `AfterEach` (or maybe even in both `BeforeEach` and `AfterEach`), because we will leave the `object` with state set and this object may be used in another test class.